crates/forge_services/src/tool_services/fs_create.rs

```diff
@@ -46,15 +46,23 @@ impl<F: FileDirectoryInfra + FileInfoInfra + FileReaderInfra + FileWriterInfra +
         // Check if the file exists
         let file_exists = self.0.is_file(path).await?;
 
-        // If file exists and overwrite flag is not set, return an error with the
-        // existing content
+        // If file exists and overwrite flag is not set, create a temporary file
+        // and return an error with helpful next steps
         if file_exists && !overwrite {
-            // Special message for the LLM
-            return Err(anyhow::anyhow!(
-                "Cannot overwrite existing file: overwrite flag not set.",
-            ))
-            // What the user sees
-            .with_context(|| format!("File already exists at {}", path.display()));
+            // Create temporary file path
+            let tmp_path_str = format!("{}.tmp", path.display());
+            let tmp_path = Path::new(&tmp_path_str);
+
+            // Write content to temporary file
+            self.0
+                .write(tmp_path, Bytes::from(content), capture_snapshot)
+                .await?;
+
+            // Return the specific error that will be handled at the operation level
+            return Ok(FsCreateOutput::Failure {
+                original_path: path.display().to_string(),
+                temp_file_path: tmp_path.display().to_string(),
+            });
```

Comment:
Writing to temp file can be moved to tool_executor. Service should not automatically assume that it needs to write in tmp file.